lib/led/rgb.js

```diff
@@ -331,6 +336,61 @@ class RGB {
     return this[this.isOn ? "off" : "on"]();
   }
 
+  /**
+   * pulse Fade the RGB in and out in a loop with specified time
+   * @param  {number} duration Time in ms that a fade in/out will elapse
+   * @return {RGB}
+   *
+   * - or -
+   *
+   * @param  {Object} val An Animation() segment config object
+   */
+
+  // rgb pulse function similar to led pulse function
+  // uses state.prev for the current color
+  pulse(duration, callback) {
+    const state = priv.get(this);
+    var currentColor = state.prev;
```

Comment:
`pulse()` uses the last color set using `color()`, saved to `state.prev` instead of checking the actual current color of the RGB as done in the prior implementation of `pulse()`. This is to avoid a problem where subsequent calls to `pulse` would lead to smaller and smaller intensity ranges. @islemaster gives an example of this case [here](https://github.com/code-dot-org/johnny-five/pull/10/files#r160807057).